src/react-apps/applications/runtime/src/components/base/FileUploadComponent.tsx

```diff
@@ -1,45 +1,332 @@
 import * as React from 'react';
+import DropZone from 'react-dropzone';
+import { connect } from 'react-redux';
+import uuid = require('uuid');
+import altinnTheme from '../../../../shared/src/theme/altinnStudioTheme';
+import { getLanguageFromKey } from '../../../../shared/src/utils/language';
+import FormFillerActionDispatchers from '../../actions/formFillerActions/formFillerActionDispatcher';
+import '../../styles/FileUploadComponent.css';
+import { renderValidationMessagesForComponent } from '../../utils/render';
 
-export interface IFileUploadProps {
+export interface IFileUploadProvidedProps {
   id: string;
-  component: IFormComponent;
-  formData: any;
-  handleDataChange: (value: any) => void;
+  component: IFormFileUploaderComponent;
   isValid?: boolean;
+  validationMessages?: IComponentValidations;
+  language: any;
+}
+
+export interface IFileUploadProps extends IFileUploadProvidedProps {
+  attachments: IAttachment[];
 }
 
 export interface IFileUploadState {
   title: string;
   component: string;
   description: string;
+  attachments: IAttachment[];
+  showFileUpload: boolean;
+  validations: string[];
 }
 
-export class FileUploadComponent
+const baseStyle = {
+  width: 'auto',
+  height: '15.6rem',
+  borderWidth: '2px',
+  borderColor: altinnTheme.altinnPalette.primary.blue,
+  borderStyle: 'dashed',
+  cursor: 'pointer',
+};
+const activeStyle = {
+  borderStyle: 'solid',
+};
+const rejectStyle = {
+  borderStyle: 'solid',
+  borderColor: altinnTheme.altinnPalette.primary.red,
+};
+const validationErrorStyle = {
+  borderStyle: 'dashed',
+  borderColor: altinnTheme.altinnPalette.primary.red,
+};
+
+export class FileUploadComponentClass
   extends React.Component<IFileUploadProps, IFileUploadState> {
 
-  public onDataChanged = (e: any) => {
-    this.props.handleDataChange(e.target.value);
+  public static getDerivedStateFromProps(props: IFileUploadProps, state: IFileUploadState): IFileUploadState {
```

Comment:
Do you need this `getDerivedStateFromProps`? Since all you do is to set attachment to the state. Isn't it enough to have it in the constructor?

Reply:
Yes, since attachments are connected at this component, it does not trigger the constructor when redux store is updated, only an update.